### DISCUSSION TITLE
How to use incremental mode with delete DML from mariadb to Oracle ?

### DISCUSSION THREAD

**Discussion by ryantle** [osalvador/ReplicaDB#94] [General]
How to use incremental mode with delete DML from mariadb to Oracle ?

Replica Db now support Delete Operation or not ?

please recommend.

**Comment by osalvador** [comment]
Hi @ryantle,

If you want to DELETE the sink table, just use the `complete` mode. The `incremental` mode aims to replicate only new rows and MERGE those rows in the sink table. 

If you only want to replicate new rows by deleting old rows in the sink table, use the `complete` mode with the custom parameters `source.query` or `source.where`.

**Reply by ryantle** [reply]
hi @osalvador 
  it's mean now Replicadb incremental mode , working with update and insert not work with delete. 

how about table need data up to date near real time (insert, update, delete), Do you recommend some product that can cdc all DML in near real time and simple to use like replicadb?

thank you for your answer.

**Reply by osalvador** [reply]
Hi @ryantle ,

OK, sorry, I didn't understand your question.

To get DELETE events in a database, you need a CDC tool. CDC tools are usually complex and expensive. Recently I was working with [Debezium](https://debezium.io/) an open source distributed platform for change data capture. It works very well, but requires more infrastructure that is beyond the scope of replicadb.

Thanks to you!
